### DISCUSSION TITLE
hot plugging and unplugging of the video signal

### DISCUSSION THREAD

**Discussion by pohsnug** [Roger-random/ESP_8_BIT_composite#57] [General]
When the video signal is disconnected from pin 25 and reconnected, the monitor does not automatically pick up the image. The video signal is detected by the monitor only after turning the monitor off and on again. Is it possible to fix this with a circuit solution or in the code? On a similar circuit with Arduino and the TVout.h library, hot plugging and unplugging of the signal works, so I conclude that it is theoretically possible.
Would be very grateful, this help is for a good cause!

**Comment by Roger-random** [comment]
I am unable to duplicate the issue here. If I hot unplug/plug my composite cable, my video signal resumes as expected. Could you try the experiment with a different ESP32 and/or a different TV? This may be an issue with your hardware.

Since I could not replicate the issue, I have no way of diagnosing the code or to verify any changes.

**Comment by pohsnug** [comment]
Thank you for the quick response! Good news for me — this should work. Bad news — I've been struggling with the solution for a week, but still no luck... 
I'm testing on two different ESP32s. I've tried different power sources for the monitors and microcontrollers (power supplies and batteries). I can't switch monitors since I plan to use these exact ones.
In the Arduino+TVout.h+this monitor setup, hot plugging/unplugging works, but the library doesn't suit me due to its limited functionality.
Another strange thing: the monitor has two video inputs (white and yellow, with white being prioritized if there is a video signal on both). The ESP32+your library only works through the yellow input. The white one doesn't pick up the signal at all, even when it's connected solo. On Arduino, both inputs work (and both after hot swapping).
I've exhausted all options, so I had to reach out to you. The library is really awesome, especially in projects where a monitor is needed to display video signals from another source (in my case, a video receiver RX5808). Now I just need to find how to switch between the two video signals.
![5427236631208912825](https://github.com/user-attachments/assets/425dbff3-5fa3-4bfa-9f58-2c847e1b8701)
![5427236631208912826](https://github.com/user-attachments/assets/12849059-d4ee-4bfb-8e8d-7ea40fa4f606)

**Comment by Roger-random** [comment]
I understand you wish to use these specific screens, but trying a different screen is useful for gathering data. If you try with a different screen and the problem does not occur, then you would know the problem is specific to these screens. If the problem still occurs, you know the screens are innocent.

On the software side you could also try FabGL and LovyanGFX to see if the same issue occurs with those libraries. https://github.com/Roger-random/ESP_8_BIT_composite?tab=readme-ov-file#alternatives

**Comment by pohsnug** [comment]
So, I tried FabGL. And it works! Same ESP, same display, test code from the library, and hot switching works. I'm not sure if I'll be able to implement my graphics on this library. It seems more complex compared to esp_8_bit... 
I compared the video signals on the oscilloscope. The frequency changes actively, so it's not important. But the Vmax and RMS differ consistently, with FabGL having a slightly higher amplitude. I’m not sure if this could affect signal detection by the monitor, but it's the only difference I found. The readings for Arduino are similar to FabGL. 
Anyway, thanks for your help! Now I’ll figure out what to do next. I already have the graphics done on esp_8_bit and I like it...
![5431423421163628446](https://github.com/user-attachments/assets/84f41218-61e3-4a3c-868e-b8b5c0ff5c1e)
![5431423421163628444](https://github.com/user-attachments/assets/31e1554f-4b63-4d0e-bd03-db502934fcc6)
![5431423421163628426](https://github.com/user-attachments/assets/c43dfad5-7b95-4ca1-84c5-4a0f0577bc65)

**Comment by Roger-random** [comment]
I'm happy to hear of your success!
